keanu-project/src/main/java/io/improbable/keanu/tensor/TensorShape.java

```diff
@@ -215,5 +217,16 @@ public static boolean isScalar(int[] shape) {
         return highRankCopy;
     }
 
+    /**
+     * Removes a dimension from a shape
+     *
+     * @param shape the shape to remove the dimension from
+     * @param dimension the dimension to remove
+     * @return the shape without the given dimension
+     */
+    public static int[] argMaxShape(int[] shape, int dimension) {
```

Comment:
This is more of an argMax concern than a TensorShape concern. I'd either move this to the class it's needed in or make a more general purpose method like @GeorgeNash is suggesting. However, since it's really just a wrapper on `ArrayUtils.remove(shape, dimension)`, which is a single line, I'd say it might not be worth the coupling/abstraction it causes.

Reply:
Turns out this isn't even something we need to do if we throw out the weird rank 1 behaviour, so I'm throwing this function out.

Reply:
I stand corrected, it is something we need to keep! The logic is slightly more complex so I'll keep it here in the more generic form

Reply:
Actually I do need to implement this, I'll go with the more general version as the logic is more complicated now.